crates/turborepo-lib/src/config.rs

```diff
@@ -0,0 +1,132 @@
+use std::path::{Path, PathBuf};
+
+use anyhow::Result;
+use config::{Config, Environment};
+// Go's xdg implementation uses FOLDERID_LocalAppData for config home
+// https://github.com/adrg/xdg/blob/master/paths_windows.go#L28
+// Rust xdg implementations uses FOLDERID_RoamingAppData for config home
+// We use cache_dir so we can find the config dir that the Go code uses
+#[cfg(windows)]
+use dirs_next::data_local_dir as config_dir;
+#[cfg(not(windows))]
+use dirs_next::config_dir;
+use serde::{Deserialize, Serialize};
+
+pub fn default_user_config_path() -> Result<PathBuf> {
+    config_dir().map(|p| p.join("turborepo").join("config.json")).context("default config path not found")
+}
+
+#[allow(dead_code)]
+pub fn data_dir() -> Option<PathBuf> {
+    dirs_next::data_dir().map(|p| p.join("turborepo"))
+}
+
+// Inner struct that matches the config file schema
+#[derive(Debug, Deserialize, Serialize, Clone, PartialEq, Eq, Default)]
+struct UserConfigInner {
+    token: Option<String>,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct UserConfig {
```

Comment:
Could we add a comment explaining why we have a `disk_config` and `config`? I'm a little confused as to what's going on.